src/plugins/shared_ux/public/components/redirect_app_links/redirect_app_links.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FunctionComponent, useRef, useMemo } from 'react';
+import useObservable from 'react-use/lib/useObservable';
+import { ApplicationStart } from 'src/core/public';
+import { createNavigateToUrlClickHandler } from './click_handler';
+
+export interface RedirectAppLinksProps extends React.HTMLAttributes<HTMLDivElement> {
+  navigateToUrl: ApplicationStart['navigateToUrl'];
+  currentAppId$: ApplicationStart['currentAppId$'];
+  className?: string;
+  'data-test-subj'?: string;
+}
+
+/**
+ * Utility component that will intercept click events on children anchor (`<a>`) elements to call
+ * `application.navigateToUrl` with the link's href. This will trigger SPA friendly navigation
+ * when the link points to a valid Kibana app.
+ *
+ * @example
+ * ```tsx
+ * <RedirectAppLinks application={application}>
+ *   <a href="/base-path/app/another-app/some-path">Go to another-app</a>
+ * </RedirectAppLinks>
+ * ```
+ *
+ * @remarks
+ * It is recommended to use the component at the highest possible level of the component tree that would
+ * require to handle the links. A good practice is to consider it as a context provider and to use it
+ * at the root level of an application or of the page that require the feature.
+ */
+export const RedirectAppLinks: FunctionComponent<RedirectAppLinksProps> = ({
+  navigateToUrl,
+  currentAppId$,
+  children,
+  ...otherProps
+}) => {
+  const currentAppId = useObservable(currentAppId$, undefined);
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  const clickHandler = useMemo(
+    () =>
+      containerRef.current && currentAppId
+        ? createNavigateToUrlClickHandler({
+            container: containerRef.current,
+            navigateToUrl,
+          })
+        : undefined,
+    [currentAppId, navigateToUrl]
+  );
+
+  return (
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
```

Comment:
Need a comment why this is necessary.  @cchaos do you have any concerns about this?

Reply:
I always have concerns about adding `onClick` behavior to typically non-interactive components, but it looks like this is just to intercept any child `href`s. Other than that, I don't really know what this component is supposed to accomplish.

Reply:
Hope this helps a bit: https://github.com/elastic/kibana/blob/main/src/plugins/kibana_react/public/page_template/no_data_page/no_data_card/elastic_agent_card.tsx#L74